.github/workflows/checks.yaml

```diff
@@ -70,7 +70,7 @@ jobs:
           args: --out-format=colored-line-number
       - if: matrix.directory == 'service'
         run: .github/scripts/init-temp-keys.sh
-      - run: go test ./... -short
+      - run: go test ./... -short -fuzztime=30s
```

Comment:
I don't think we want to introduce fuzzing as part of our normal unit test workflow (beyond the fuzz seeds, which will be validated without any changes here).

We should keep unit tests running fast and be deterministic.  Unfortunately fuzzing is neither, it's very easy to have one time failures and it often needs to run for a long period to be valuable.

I would like to introduce regular automated fuzzing, but I think we should do that as future work.  When we do introduce it we need to consider:
* How do we securely report the failed test payload, since it may not be easy to recreate (or even impossible)
* What cadence makes sense (cost vs benefit balance)

Since this project is open source one option we may consider to meet these needs: https://github.com/google/oss-fuzz

But in any case, I rather do these changes separate from the fix here.